Automattic/wp-calypso
# Modularise products list state

This PR is one of many working on modularising state in Calypso. This one handles products list state.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

Fixes #42470.

#### Changes proposed in this Pull Request

* Modularise products list state
* Split up products list selectors into individual modules

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/read` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and note that there's no `productsList` key, even if you expand the list of keys. This indicates that the products list state hasn't been loaded yet.
* Click `My Sites` on the top bar.
* Verify that a `productsList` key is added with the products list state. This indicates that the products list state has now been loaded.
* Verify that products list-related functionality works normally. This indicates that I didn't mess up.
